{"url": "https://api.github.com/repos/datalad/datalad/issues/1112", "repository_url": "https://api.github.com/repos/datalad/datalad", "labels_url": "https://api.github.com/repos/datalad/datalad/issues/1112/labels{/name}", "comments_url": "https://api.github.com/repos/datalad/datalad/issues/1112/comments", "events_url": "https://api.github.com/repos/datalad/datalad/issues/1112/events", "html_url": "https://github.com/datalad/datalad/issues/1112", "id": 188732814, "node_id": "MDU6SXNzdWUxODg3MzI4MTQ=", "number": 1112, "title": "AutoMagicIO not working (permission issues?)", "user": {"login": "mih", "id": 136479, "node_id": "MDQ6VXNlcjEzNjQ3OQ==", "avatar_url": "https://avatars.githubusercontent.com/u/136479?v=4", "gravatar_id": "", "url": "https://api.github.com/users/mih", "html_url": "https://github.com/mih", "followers_url": "https://api.github.com/users/mih/followers", "following_url": "https://api.github.com/users/mih/following{/other_user}", "gists_url": "https://api.github.com/users/mih/gists{/gist_id}", "starred_url": "https://api.github.com/users/mih/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/mih/subscriptions", "organizations_url": "https://api.github.com/users/mih/orgs", "repos_url": "https://api.github.com/users/mih/repos", "events_url": "https://api.github.com/users/mih/events{/privacy}", "received_events_url": "https://api.github.com/users/mih/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 63778590, "node_id": "MDU6TGFiZWw2Mzc3ODU5MA==", "url": "https://api.github.com/repos/datalad/datalad/labels/severity-normal", "name": "severity-normal", "color": "ff8888", "default": false, "description": "standard severity"}], "state": "closed", "locked": false, "assignee": {"login": "yarikoptic", "id": 39889, "node_id": "MDQ6VXNlcjM5ODg5", "avatar_url": "https://avatars.githubusercontent.com/u/39889?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yarikoptic", "html_url": "https://github.com/yarikoptic", "followers_url": "https://api.github.com/users/yarikoptic/followers", "following_url": "https://api.github.com/users/yarikoptic/following{/other_user}", "gists_url": "https://api.github.com/users/yarikoptic/gists{/gist_id}", "starred_url": "https://api.github.com/users/yarikoptic/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yarikoptic/subscriptions", "organizations_url": "https://api.github.com/users/yarikoptic/orgs", "repos_url": "https://api.github.com/users/yarikoptic/repos", "events_url": "https://api.github.com/users/yarikoptic/events{/privacy}", "received_events_url": "https://api.github.com/users/yarikoptic/received_events", "type": "User", "site_admin": false}, "assignees": [{"login": "yarikoptic", "id": 39889, "node_id": "MDQ6VXNlcjM5ODg5", "avatar_url": "https://avatars.githubusercontent.com/u/39889?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yarikoptic", "html_url": "https://github.com/yarikoptic", "followers_url": "https://api.github.com/users/yarikoptic/followers", "following_url": "https://api.github.com/users/yarikoptic/following{/other_user}", "gists_url": "https://api.github.com/users/yarikoptic/gists{/gist_id}", "starred_url": "https://api.github.com/users/yarikoptic/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yarikoptic/subscriptions", "organizations_url": "https://api.github.com/users/yarikoptic/orgs", "repos_url": "https://api.github.com/users/yarikoptic/repos", "events_url": "https://api.github.com/users/yarikoptic/events{/privacy}", "received_events_url": "https://api.github.com/users/yarikoptic/received_events", "type": "User", "site_admin": false}], "milestone": {"url": "https://api.github.com/repos/datalad/datalad/milestones/37", "html_url": "https://github.com/datalad/datalad/milestone/37", "labels_url": "https://api.github.com/repos/datalad/datalad/milestones/37/labels", "id": 2123515, "node_id": "MDk6TWlsZXN0b25lMjEyMzUxNQ==", "number": 37, "title": "Release 0.4.2", "description": "was never released -- will be absorbed into 0.5", "creator": {"login": "yarikoptic", "id": 39889, "node_id": "MDQ6VXNlcjM5ODg5", "avatar_url": "https://avatars.githubusercontent.com/u/39889?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yarikoptic", "html_url": "https://github.com/yarikoptic", "followers_url": "https://api.github.com/users/yarikoptic/followers", "following_url": "https://api.github.com/users/yarikoptic/following{/other_user}", "gists_url": "https://api.github.com/users/yarikoptic/gists{/gist_id}", "starred_url": "https://api.github.com/users/yarikoptic/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yarikoptic/subscriptions", "organizations_url": "https://api.github.com/users/yarikoptic/orgs", "repos_url": "https://api.github.com/users/yarikoptic/repos", "events_url": "https://api.github.com/users/yarikoptic/events{/privacy}", "received_events_url": "https://api.github.com/users/yarikoptic/received_events", "type": "User", "site_admin": false}, "open_issues": 0, "closed_issues": 12, "state": "closed", "created_at": "2016-11-09T01:12:36Z", "updated_at": "2017-02-09T16:13:36Z", "due_on": null, "closed_at": "2017-02-09T16:13:13Z"}, "comments": 4, "created_at": "2016-11-11T10:44:31Z", "updated_at": "2016-11-12T17:09:26Z", "closed_at": "2016-11-12T17:09:16Z", "author_association": "MEMBER", "active_lock_reason": null, "body": "This command would fail due to missing annex'ed files:\r\n````\r\n$ python code/mk_ds.py 16 stupid                        \r\nTraceback (most recent call last):\r\n...\r\n  File \"/usr/lib/python2.7/dist-packages/nibabel/loadsave.py\", line 40, in load\r\n    raise FileNotFoundError(\"No such file: '%s'\" % filename)\r\nnibabel.py3k.FileNotFoundError: No such file: 'src/templatesntransforms/sub-16/spaghetti1/mask.nii.gz'\r\n````\r\n\r\nAutoMagicIO should come to the rescue...but doesn't:\r\n\r\n````\r\n$ python -m datalad code/mk_ds.py 16 stupid           \r\n[INFO   ] Activating DataLad's AutoMagicIO \r\n[INFO   ] Running code of code/mk_ds.py \r\n[ERROR  ] <class 'git.exc.InvalidGitRepositoryError'>: /etc \r\n[WARNING] Failed proxying open with ('/home/mih/datalad/datalad.egg-info/PKG-INFO', 'rb'), {}: [Errno 13] Permission denied: '/home/mih/datalad/.git/config.lock' [util.py:_obtain_lock_or_raise:758] \r\n[WARNING] Failed proxying open with ('/home/mih/datalad/datalad.egg-info/PKG-INFO', 'rb'), {}: [Errno 13] Permission denied: '/home/mih/datalad/.git/config.lock' [util.py:_obtain_lock_or_raise:758] \r\nfatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz\r\nfatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz\r\nfatal: Not a valid object name src/aligned/sub-16/in_spaghetti1/sub-16_task-taste_run-1_bold.nii.gz\r\nfatal: Not a valid object name src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz\r\nfatal: Not a valid object name src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz\r\nTraceback (most recent call last):\r\n  File \"/usr/lib/python2.7/runpy.py\", line 162, in _run_module_as_main\r\n    \"__main__\", fname, loader, pkg_name)\r\n  File \"/usr/lib/python2.7/runpy.py\", line 72, in _run_code\r\n    exec code in run_globals\r\n  File \"/home/mih/datalad/datalad/__main__.py\", line 102, in <module>\r\n    main()\r\n  File \"/home/mih/datalad/datalad/__main__.py\", line 93, in main\r\n    runctx(code, globs, globs)\r\n  File \"/home/mih/datalad/datalad/__main__.py\", line 41, in runctx\r\n    exec(cmd, globals, locals)\r\n  File \"code/mk_ds.py\", line 42, in <module>\r\n    runs = [load_run(subid, runid) for runid in ('1', '2', '3', '4')]\r\n  File \"code/mk_ds.py\", line 19, in load_run\r\n    ds = fmri_dataset(bold_fname, mask=mask_fname)\r\n  File \"/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py\", line 258, in fmri_dataset\r\n    maskimg = _load_anyimg(mask)\r\n  File \"/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py\", line 396, in _load_anyimg\r\n    data = _img2data(src)\r\n  File \"/usr/lib/python2.7/dist-packages/mvpa2/datasets/mri.py\", line 72, in _img2data\r\n    img = nibabel.load(src)\r\n  File \"/usr/lib/python2.7/dist-packages/nibabel/loadsave.py\", line 48, in load\r\n    filename)\r\nnibabel.filebasedimages.ImageFileError: Cannot work out file type of \"src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz\"\r\n````\r\n\r\nThis is using a datalad that was installed by different user as the one running the command. But write permissions shouldn't be necessary, right?\r\n\r\nFTR: A manual\r\n\r\n ````\r\n$ datalad get src/templatesntransforms/sub-16/spaghetti1/mask_insula.nii.gz\r\n````\r\n\r\nworks as expected.", "closed_by": {"login": "yarikoptic", "id": 39889, "node_id": "MDQ6VXNlcjM5ODg5", "avatar_url": "https://avatars.githubusercontent.com/u/39889?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yarikoptic", "html_url": "https://github.com/yarikoptic", "followers_url": "https://api.github.com/users/yarikoptic/followers", "following_url": "https://api.github.com/users/yarikoptic/following{/other_user}", "gists_url": "https://api.github.com/users/yarikoptic/gists{/gist_id}", "starred_url": "https://api.github.com/users/yarikoptic/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yarikoptic/subscriptions", "organizations_url": "https://api.github.com/users/yarikoptic/orgs", "repos_url": "https://api.github.com/users/yarikoptic/repos", "events_url": "https://api.github.com/users/yarikoptic/events{/privacy}", "received_events_url": "https://api.github.com/users/yarikoptic/received_events", "type": "User", "site_admin": false}, "reactions": {"url": "https://api.github.com/repos/datalad/datalad/issues/1112/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/datalad/datalad/issues/1112/timeline", "performed_via_github_app": null}